statsmodels/base/distributed_estimation.py

```diff
@@ -0,0 +1,677 @@
+    pnum : scalar
+        Index of current partition
+    partitions : scalar
+        Total number of partitions
+    fit_kwds : dict-like or None
+        Keyword arguments to be given to fit_regularized
+
+    Returns
+    -------
+    An array of the paramters for the regularized fit
+    """
+
+    if fit_kwds is None:
+        raise ValueError("_est_regularized_naive currently " +
+                         "requires that fit_kwds not be None.")
+
+    return mod.fit_regularized(**fit_kwds).params
+
+
+def _est_unregularized_naive(mod, pnum, partitions, fit_kwds=None):
+    """estimates the unregularized fitted parameters.
+
+    Parameters
+    ----------
+    mod : statsmodels model class instance
+        The model for the current partition.
+    pnum : scalar
+        Index of current partition
+    partitions : scalar
+        Total number of partitions
+    fit_kwds : dict-like or None
+        Keyword arguments to be given to fit
+
+    Returns
+    -------
+    An array of the parameters for the fit
+    """
+
+    if fit_kwds is None:
+        raise ValueError("_est_unregularized_naive currently " +
+                         "requires that fit_kwds not be None.")
+
+    return mod.fit(**fit_kwds).params
+
+
+def _join_naive(params_l, threshold=0):
+    """joins the results from each run of _est_<type>_naive
+    and returns the mean estimate of the coefficients
+
+    Parameters
+    ----------
+    params_l : list
+        A list of arrays of coefficients.
+    threshold : scalar
+        The threshold at which the coefficients will be cut.
+    """
+
+    p = len(params_l[0])
+    partitions = len(params_l)
+
+    params_mn = np.zeros(p)
+    for params in params_l:
+        params_mn += params
+    params_mn /= partitions
+
+    params_mn[np.abs(params_mn) < threshold] = 0
+
+    return params_mn
+
+
+def _calc_grad(mod, params, alpha, L1_wt, score_kwds):
+    """calculates the log-likelihood gradient for the debiasing
+
+    Parameters
+    ----------
+    mod : statsmodels model class instance
+        The model for the current partition.
+    params : array-like
+        The estimated coefficients for the current partition.
+    alpha : scalar or array-like
+        The penalty weight.  If a scalar, the same penalty weight
+        applies to all variables in the model.  If a vector, it
+        must have the same length as `params`, and contains a
+        penalty weight for each coefficient.
+    L1_wt : scalar
+        The fraction of the penalty given to the L1 penalty term.
+        Must be between 0 and 1 (inclusive).  If 0, the fit is
+        a ridge fit, if 1 it is a lasso fit.
+    score_kwds : dict-like or None
+        Keyword arguments for the score function.
+
+    Returns
+    -------
+    An array-like object of the same dimension as params
+
+    Notes
+    -----
+    In general:
+
+    gradient l_k(params)
+
+    where k corresponds to the index of the partition
+
+    For the simple linear case:
```

Comment:
It's not really "simple linear" if there are more than 1 covariates, better to use "OLS" I think.